AIFG-3D-Connect-4/Grid.cpp

```diff
@@ -0,0 +1,81 @@
+#include "Grid.h"
+
+Grid::Grid(int t_gridIndex)
+{
+	m_gridIndex = t_gridIndex;
+	sf::RectangleShape rect = sf::RectangleShape(sf::Vector2f(static_cast<float>(m_PIECE_SIZE), static_cast<float>(m_PIECE_SIZE)));
+	for (Coordinate::Type col = 0u; col < GameBoard::SIZE; col++)
+	{
+		for (Coordinate::Type row = 0u; row < GameBoard::SIZE; row++)
+		{
+			setUpRect(sf::Color::Black, rect, row, col);
+		}
+	}
+}
+
+Grid::~Grid()
+{
+
+}
+
+void Grid::render(sf::RenderWindow& t_window)
+{
+	sf::CircleShape piece = sf::CircleShape(m_PIECE_SIZE / 2, m_PIECE_SIZE / 2);
+	for (Coordinate::Type col = 0u; col < GameBoard::SIZE; col++)
+	{
+		for (Coordinate::Type row = 0u; row < GameBoard::SIZE; row++)
+		{
+			PieceType type = m_gameBoard->getPiece({ row, static_cast<Coordinate::Type>(m_gridIndex), col });
+			if (type == PieceType::None) { continue; }
+			else if (type == PieceType::Red)
+			{
+				setUpPiece(sf::Color::Red, piece, row, col);
```

Comment:
If the pieces are being added to a vector, should this be called every render? Would the size of the vector not be forever growing?